### PR TITLE
Add evaluator

### PR DESCRIPTION
**Pull request by fcogidi**
# Feature

## Short Description
- Refactor `cyclops.models` package.
- Add `MetricCollection` class.
- Add `Evaluator` class.

## Tests Added
tests/cyclops/evaluate/metrics/test_metric_collection.py

### CODE REVIEW

**Comment by fcogidi** [comment]
@a-kore Changes to the `cyclops.models` package may break some functionalities in the `cyclops.monitor` package. Please review.

**Review by a-kore — APPROVED** [review]
It looks okay, all test are still passing for the drift modules.

**Comment by fcogidi** [comment]
> It looks okay, all test are still passing for the drift modules.

An example of something that may not work as expected is this method: https://github.com/VectorInstitute/cyclops/blob/3dfa11662708bb0564bbd1db1c0cd706f02d0465/cyclops/monitor/utils.py#L183
The fit API has changed to `fit(X, y=None, **fit_params)`. Also, the `SKModel` wrapper will not do anything with the `save_path` parameter from the constructor. It's now passed along with a call to `save_model(filepath, overwrite=True, **kwargs)`.

Also, the way config files are structured has changed, especially for PyTorch models.

**Comment by a-kore** [comment]
I see.  The core modules for drift just use the models from the model library and it all seems to still work.  There isn't any test coverage for the `retrainers` module yet which would use the fit API.  So, I can build around those changes once I write those tests.

File: cyclops/evaluate/metrics/precision_recall_curve.py [hunk 1]
```diff
@@ -328,108 +332,113 @@ def compute(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
...
-        ...         [[0.1, 0.9], [0.8, 0.2]]]
-        >>> for t, p in zip(target, preds):
-        ...     metric.update_state(t, p)
-        >>> metric.compute()
-        (array([[0.5, 0.5, 0. , 1. ],
-                [0.5, 0.5, 0. , 1. ]]),
-        array([[1. , 0.5, 0. , 0. ],
-                [1. , 0.5, 0. , 0. ]]),
-        array([0. , 0.5, 1. ]))
+    Examples
```
Comment (amrit110) [inline cyclops/evaluate/metrics/precision_recall_curve.py:381]: I think you can only have one `Examples` section, but then use (binary), (multiclass) and line spacing to separate those. Right now, with multiple `Examples` mentioned, sphinx gets confused and doesn't render them.

**Review by amrit110 — CHANGES_REQUESTED** [review]

File: cyclops/evaluate/metrics/precision_recall_curve.py (hunk 1 above)
Comment (fcogidi) [reply cyclops/evaluate/metrics/precision_recall_curve.py]: There are still some errors about indentation, but I don't know how to fix those.

Comment (amrit110) [reply cyclops/evaluate/metrics/precision_recall_curve.py]: That's fine, can fix those later!

Comment (fcogidi) [reply cyclops/evaluate/metrics/precision_recall_curve.py]: Do you know which indentation it's complaining about though?

Comment (amrit110) [reply cyclops/evaluate/metrics/precision_recall_curve.py]: Hmm I recall fixing some of those a while back. These don't seem to affect the docs rendering though. I'll investigate them in a bit. Have added to my TODOs.

**Pull request merged**